firestore/query.go

```diff
@@ -1083,8 +1304,16 @@ func newQueryDocumentIterator(ctx context.Context, q *Query, tid []byte, rs *rea
 	}
 }
 
-func (it *queryDocumentIterator) next() (_ *DocumentSnapshot, err error) {
+// opts override the options stored in it.q.runQuerySettings
+func (it *queryDocumentIterator) next(opts ...RunOption) (_ *DocumentSnapshot, err error) {
```

Comment:
Does this mean we support changing options mid-query?

Reply:
This is an unexported method. User cannot call this directly

But yes, user can definitely do this:
:
```
        // Analyze true
	docIter := client.Collection("cities").
		WithRunOptions(firestore.ExplainOptions{Analyze: true}).
		Documents(ctx)

         // Read single document
         docSnapshot, _ := docIter.Next()
         
         // Read rest of the documents with Analyze false
         res, _ := docIter.GetAllWithOptions(firestore.ExplainOptions{Analyze: true})
```

Reply:
Why would anyone do that?

Reply:
design has been updated. User can no longer do above